custom/icds_reports/utils/data_pull.sql

```diff
@@ -0,0 +1,79 @@
+CREATE TABLE "temp_visit_table" AS (
+    SELECT
+        location_id as supervisor_id,
+        COUNT (DISTINCT awc_id) as unique_visits
+    FROM "ucr_icds-cas_static-awc_mgt_forms_ad1b11f0"
+    WHERE
+        submitted_on>= '2020-03-01'
+        AND submitted_on<'2020-05-01'
+        AND location_entered IS NOT NULL
+        AND location_entered <> ''
+    GROUP BY location_id
+);
+
+--                                                                                           QUERY PLAN
+-- ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+--  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
+--    Group Key: remote_scan.supervisor_id
+--    ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+--          Task Count: 64
+--          Tasks Shown: One of 64
+--          ->  Task
+--                Node: host=100.71.184.232 port=6432 dbname=icds_ucr
+--                ->  GroupAggregate  (cost=1205.95..1235.94 rows=251 width=41)
+--                      Group Key: location_id
+--                      ->  Sort  (cost=1205.95..1215.11 rows=3664 width=66)
+--                            Sort Key: location_id
+--                            ->  Seq Scan on "ucr_icds-cas_static-awc_mgt_forms_ad1b11f0_103162" "ucr_icds-cas_static-awc_mgt_forms_ad1b11f0"  (cost=0.00..989.06 rows=3664 width=66)
+--                                  Filter: ((location_entered IS NOT NULL) AND (submitted_on >= '2020-02-01'::date) AND (submitted_on < '2020-04-01'::date) AND (location_entered <> ''::text))
+-- (13 rows)
+
+
+
+
+SELECT
+    t.state_name,
+    t.supervisor_name,
+    t.supervisor_site_code,
+    CASE WHEN al.num_supervisor_launched>0 THEN 'Launched' ELSE 'UnLaunched' END as supervisor_launched_status,
+    awc.num_awcs as total_awcs,
+    ucr.unique_visits as visited_awcs,
+    CASE
+        WHEN awc.num_awcs = ucr.unique_visits THEN 'YES'
+        ELSE 'NO'
+    END as all_visited
+    FROM "agg_ls" al
+    LEFT JOIN "temp_visit_table" ucr
+        ON awc.supervisor_id = ucr.supervisor_id
+    LEFT JOIN "awc_location_local" t
+        ON (t.supervisor_id = awc.supervisor_id
+        AND t.aggregation_level=awc.aggregation_level
+        AND t.aggregation_level=4)
+    LEFT JOIN "agg_awc_2020-04-01_4" awc
+        ON (
+            awc.supervisor_id = al.supervisor_id
+            AND al.aggregation_level=awc.aggregation_level AND awc.aggregation_level=4 AND awc.month='2020-04-01')
+    WHERE al.state_is_test<>1 AND al.supervisor_is_test<>1;
```

Comment:
you will need to put agg_level=4 in the where clause as well to finally exclude all other levels from final result

Reply:
updated